local/compose.go

```diff
@@ -175,40 +176,151 @@ func (s *composeService) Up(ctx context.Context, project *types.Project, detach
 	if err != nil {
 		return err
 	}
-	for k, network := range project.Networks {
-		if !network.External.External && network.Name == k {
-			network.Name = fmt.Sprintf("%s_%s", project.Name, k)
-			project.Networks[k] = network
-		}
-		network.Labels = network.Labels.Add(networkLabel, k)
-		network.Labels = network.Labels.Add(projectLabel, project.Name)
-		network.Labels = network.Labels.Add(versionLabel, ComposeVersion)
-		err := s.ensureNetwork(ctx, network)
-		if err != nil {
-			return err
+
+	_, err = progress.Run(ctx, func(ctx context.Context) (string, error) {
+		for k, network := range project.Networks {
+			if !network.External.External && network.Name != "" {
+				network.Name = fmt.Sprintf("%s_%s", project.Name, k)
+				project.Networks[k] = network
+			}
+			network.Labels = network.Labels.Add(networkLabel, k)
+			network.Labels = network.Labels.Add(projectLabel, project.Name)
+			network.Labels = network.Labels.Add(versionLabel, ComposeVersion)
+			err := s.ensureNetwork(ctx, network)
+			if err != nil {
+				return "", err
+			}
 		}
-	}
 
-	for k, volume := range project.Volumes {
-		if !volume.External.External && volume.Name != "" {
-			volume.Name = fmt.Sprintf("%s_%s", project.Name, k)
-			project.Volumes[k] = volume
+		for k, volume := range project.Volumes {
+			if !volume.External.External && volume.Name != "" {
+				volume.Name = fmt.Sprintf("%s_%s", project.Name, k)
+				project.Volumes[k] = volume
+			}
+			volume.Labels = volume.Labels.Add(volumeLabel, k)
+			volume.Labels = volume.Labels.Add(projectLabel, project.Name)
+			volume.Labels = volume.Labels.Add(versionLabel, ComposeVersion)
+			err := s.ensureVolume(ctx, volume)
+			if err != nil {
+				return "", err
+			}
 		}
-		volume.Labels = volume.Labels.Add(volumeLabel, k)
-		volume.Labels = volume.Labels.Add(projectLabel, project.Name)
-		volume.Labels = volume.Labels.Add(versionLabel, ComposeVersion)
-		err := s.ensureVolume(ctx, volume)
+
+		err = inDependencyOrder(ctx, project, func(c context.Context, service types.ServiceConfig) error {
+			return s.ensureService(c, project, service)
+		})
 		if err != nil {
-			return err
+			return "", err
+		}
+
+		if detach {
+			err = inDependencyOrder(ctx, project, func(c context.Context, service types.ServiceConfig) error {
+				return s.startService(ctx, project, service)
+			})
+			return "", err
 		}
+
+		return "", nil
+	})
+	if detach || err != nil {
+		return err
+	}
+
+	return s.attach(ctx, project)
+}
+
+func (s *composeService) attach(ctx context.Context, project *types.Project) error {
+	consumer := formatter.NewLogConsumer(ctx, os.Stdout)
```

Comment:
See comment above, the `os.Stdout` is problematic if we call compose up over gRPC